modules/tinymce/src/core/main/ts/delete/DeleteElement.ts

```diff
@@ -87,7 +88,12 @@ const isBlock = (editor: Editor, elm: SugarElement<Node>): boolean =>
 const paddEmptyBlock = (elm: SugarElement<Node>): Optional<CaretPosition> => {
   if (Empty.isEmpty(elm)) {
     const br = SugarElement.fromHtml('<br data-mce-bogus="1">');
-    Remove.empty(elm);
+    // Remove all bogus elements except caret
+    Arr.each(Traverse.children(elm), (node) => {
+      if (node.dom.textContent !== ZWSP) {
```

Comment:
You did well to get this far without help, but `data-mce-type="format-caret"` or `id="_mce_caret"` is a better way to detect this. It's applied to the outer element in this "temporary style" mode, which is used when applying styles in a scenario where it's impossible to persist those styles to a rendered HTML document.
https://github.com/tinymce/tinymce/blob/14dd42f7015cd5573e1da14a3b725e213e3483d0/modules/tinymce/src/core/main/ts/fmt/ApplyFormat.ts#L344

There's even a method to help detect it:
https://github.com/tinymce/tinymce/blob/9dbb1e7742b0835382abc11aa2edd6c33a71f625/modules/tinymce/src/core/main/ts/fmt/CaretFormat.ts#L381

Reply:
I used `isCaretNode` rather than `isEmptyCaretFormatElement`, to avoid introducing circular dependency

Reply:
Ah I forgot that you made changes to CaretFormat so of course you knew about it.

If it's just importing constant function references, TypeScript can (much to my disgust) deal with circular dependencies.